avocado/plugins/yaml_to_mux.py

```diff
@@ -59,7 +59,51 @@ class ListOfNodeObjects(list):     # Few methods pylint: disable=R0903
     pass
 
 
-def _create_from_yaml(path, cls_node=tree.TreeNode):
+class MuxTreeNode(tree.TreeNode):
+
+    """
+    Class for bounding nodes into tree-structure with support for
+    multiplexation
+    """
+
+    def __init__(self, name='', value=None, parent=None, children=None):
+        super(MuxTreeNode, self).__init__(name, value, parent, children)
+        self.multiplex = None
+
+    def __repr__(self):
+        return 'TreeNode(name=%r)' % self.name
+
+    def merge(self, other):
+        """
+        Merges `other` node into this one without checking the name of the
+        other node. New values are appended, existing values overwritten
+        and unaffected ones are kept. Then all other node children are
+        added as children (recursively they get either appended at the end
+        or merged into existing node in the previous position.
+        """
+        super(MuxTreeNode, self).merge(other)
+        if other.multiplex is True:
```

Comment:
This is quite tricky question. The multiplex is an important tag changing the way the tree is processed. When merging two trees, we might end up in a situation where simpler tree is merged into a more-specific tree, where the simpler tree adds some default values, but the more-specific tree already contains it's variants. Then we don't want to loose the variants:

```
# simpler tree
cpus:
    count: 4
    type: x86_64

# complex tree
cpus: !mux
    x86_64:
        type: x86_64
    ppc64:
        type: ppc64

# should produce
[medic@w541 avocado ]$ avocado multiplex -m simple.yaml complex.yaml -c
Variants generated:

Variant 1:    /run/cpus/x86_64
    /run/cpus/x86_64:type => x86_64
    /run/cpus:count       => 4

Variant 2:    /run/cpus/ppc64
    /run/cpus/ppc64:type => ppc64
    /run/cpus:count      => 4
[medic@w541 avocado ]$ avocado multiplex -m complex.yaml simple.yaml -c
Variants generated:

Variant 1:    /run/cpus/x86_64
    /run/cpus/x86_64:type => x86_64
    /run/cpus:count       => 4

Variant 2:    /run/cpus/ppc64
    /run/cpus/ppc64:type => ppc64
    /run/cpus:count      => 4

# but with `self.multiplex = other.multiplex` only the value of the later is kept
[medic@w541 avocado ]$ avocado multiplex -m complex.yaml simple.yaml -c
Variants generated:

Variant 1:    /run/cpus/x86_64, /run/cpus/ppc64
    /run/cpus/ppc64:type  => ppc64
    /run/cpus/x86_64:type => x86_64
    /run/cpus:count       => 4
```